packages/event-handler/tests/unit/rest/BaseRouter.test.ts

```diff
@@ -37,6 +40,50 @@ const createTestEvent = (
   resource: '',
 });
 
+// Middleware factory functions
+const createTrackingMiddleware = (
+  name: string,
+  executionOrder: string[]
+): Middleware => {
+  return async (_params, _options, next) => {
+    executionOrder.push(`${name}-start`);
+    await next();
+    executionOrder.push(`${name}-end`);
+  };
+};
+
+const createThrowingMiddleware = (
+  name: string,
+  executionOrder: string[],
+  errorMessage: string
+): Middleware => {
+  return async (_params, _options, _next) => {
+    executionOrder.push(name);
+    throw new Error(errorMessage);
+  };
+};
+
+const createReturningMiddleware = (
+  name: string,
+  executionOrder: string[],
+  response: any
+): Middleware => {
+  return async (_params, _options, _next) => {
+    executionOrder.push(name);
+    return response;
+  };
+};
+
+const createNoNextMiddleware = (
```

Comment:
I'd consider moving these to a dedicated file under `tests/unit/rest/helpers.ts`

Reply:
Agrred